lib/middleware/checkLicenceStatus.js

```diff
@@ -2,42 +2,60 @@ const logger = require('../logger');
 const accessLicenceService = require('../services/accessLicenceService');
 const acquisitionContextService = require('../services/acquisitionContextService');
 
-
-module.exports = {
-
-    isActive :function* (req, res, next) {
-
-        try {
-
-            const currentLicence = yield accessLicenceService.getLicenceInfo(req.licenceId);
-
-            if (currentLicence.status !== 'active') {
-                logger.info({operation:'checkLicenceStatus', status:currentLicence.status});
-                const err = new Error('Licence has been revoked');
-                err.status = 403;
-                throw err;
-            }
-
-            logger.info({operation:'checkLicenceStatus', status:currentLicence.status});
-
-            res.set('Cache-Control', 'private, no-cache, no-store, must-revalidate');
-            next();
-
-        } catch (err) {
-            next(err);
+function* _updateLicenceList(req, res) {
+  let licenceList = yield accessLicenceService.getLicenceList({adminuserid:req.currentUser.uuid});
+  logger.info({operation:'updateLicenceList', licenceId:req.licenceId, isLicenceListArray: licenceList.length});
+
+  if (licenceList && licenceList.length >= 0) {
+    logger.info({operation:'updateLicenceList', licenceId:req.licenceId, isLicenceListArray: licenceList.length});
+    //if licenceList exists and has properties return the array without the current licenceId
+      licenceList = licenceList.filter((item) => {
```

Comment:
`filter` expects the callback to return `true/false`. I know technically `item` is truthy, but still this seems weird to me. :)
Also, the same behavior can be achieved w/o going through the entire array. Once we find a good value, we can stop the parsing.
Ex: 
```
licenceList.every(item => {
  if (item.licenceId !== req.licenceId) {
    req.licenceId = item.licenceId;
    return false;
  }
  return true;
});
```

Didn't really go through the entire code, but is changing the `req.licenceId` value enough? Shouldn't we redirect to the same page using this new licenceId?